---
title: An Improved Codon Modeling Approach for Accurate Estimation of the Mutation
  Bias
authors:
- T. Latrille
- N. Lartillot
date: '2022-02-01'
publishDate: '2024-11-20T20:25:54.829321Z'
publication_types:
- article-journal
publication: '*Molecular Biology and Evolution*'

abstract: Phylogenetic codon models are routinely used to characterize selective regimes
  in coding sequences. Their parametric design, however, is still a matter of debate,
  in particular concerning the question of how to account for differing nucleotide
  frequencies and substitution rates. This problem relates to the fact that nucleotide
  composition in protein-coding sequences is the result of the interactions between
  mutation and selection. In particular, because of the structure of the genetic code,
  the nucleotide composition differs between the three coding positions, with the
  third position showing a more extreme composition. Yet, phylogenetic codon models
  do not correctly capture this phenomenon and instead predict that the nucleotide
  composition should be the same for all three positions. Alternatively, some models
  allow for different nucleotide rates at the three positions, an approach conflating
  the effects of mutation and selection on nucleotide composition. In practice, it
  results in inaccurate estimation of the strength of selection. Conceptually, the
  problem comes from the fact that phylogenetic codon models do not correctly capture
  the fixation bias acting against the mutational pressure at the mutation--selection
  equilibrium. To address this problem and to more accurately identify mutation rates
  and selection strength, we present an improved codon modeling approach where the
  fixation rate is not seen as a scalar, but as a tensor. This approach gives an accurate
  representation of how mutation and selection oppose each other at equilibrium and
  yields a reliable estimate of the mutational process, while disentangling the mean
  fixation probabilities prevailing in different mutational directions.

featured: true
share: false
show_date: false
links:
- name: DOI
  url: https://doi.org/10.1093/molbev/msac005
url_pdf: '/publication/latrille-improved-2022/paper.pdf'
url_code: 'https://github.com/ThibaultLatrille/NucleotideBias'
url_dataset: ''
url_poster: ''
url_project: ''
url_slides: '/publication/latrille-improved-2022/slides.pdf'
url_source: 'https://github.com/ThibaultLatrille/PhD/tree/master/NucleotideBias'
url_video: ''

---
