---
title: Inferring Long-Term Effective Population Size with Mutation--Selection Models
authors:
- T. Latrille
- V. Lanore
- N. Lartillot
date: '2021-10-01'
publishDate: '2024-11-20T20:25:54.834537Z'
publication_types:
- article-journal
publication: '*Molecular Biology and Evolution*'

abstract: Mutation--selection phylogenetic codon models are grounded on population
  genetics first principles and represent a principled approach for investigating
  the intricate interplay between mutation, selection, and drift. In their current
  form, mutation--selection codon models are entirely characterized by the collection
  of site-specific amino-acid fitness profiles. However, thus far, they have relied
  on the assumption of a constant genetic drift, translating into a unique effective
  population size ($N_e$) across the phylogeny, clearly an unrealistic assumption. This
  assumption can be alleviated by introducing variation in $N_e$ between lineages. In
  addition to $N_e$, the mutation rate ($μ$) is susceptible to vary between lineages,
  and both should covary with life-history traits (LHTs). This suggests that the model
  should more globally account for the joint evolutionary process followed by all
  of these lineage-specific variables ($N_e$, $μ$, and LHTs). In this direction, we introduce
  an extended mutation--selection model jointly reconstructing in a Bayesian Monte
  Carlo framework the fitness landscape across sites and long-term trends in $N_e$, $μ$,
  and LHTs along the phylogeny, from an alignment of DNA coding sequences and a matrix
  of observed LHTs in extant species. The model was tested against simulated data
  and applied to empirical data in mammals, isopods, and primates. The reconstructed
  history of $N_e$ in these groups appears to correlate with LHTs or ecological variables
  in a way that suggests that the reconstruction is reasonable, at least in its global
  trends. On the other hand, the range of variation in $N_e$ inferred across species
  is surprisingly narrow. This last point suggests that some of the assumptions of
  the model, in particular concerning the assumed absence of epistatic interactions
  between sites, are potentially problematic.

featured: true
share: false
show_date: false
links:
- name: DOI
  url: https://doi.org/10.1093/molbev/msab160
url_pdf: '/publication/latrille-inferring-2021/paper.pdf'
url_code: 'https://github.com/ThibaultLatrille/MutationSelectionDrift'
url_dataset: ''
url_poster: ''
url_project: ''
url_slides: '/publication/latrille-inferring-2021/slides.pdf'
url_source: 'https://github.com/ThibaultLatrille/PhD/tree/master/MutationSelectionDrift'
url_video: ''
---
